custom_components/tuya_local/devices/hisb80a_energymeter.yaml

```diff
@@ -0,0 +1,168 @@
+name: Three Phase Meter 80A
+products:
+  - id: w5nhjnc3linrto2x
+    name: Three Phase Meter 80A
+entities:
+  - entity: switch
+    class: outlet
+    dps:
+      - id: 16
+        type: boolean
+        name: switch
+  - entity: sensor
+    name: Total Energy
+    class: energy
```

Comment:
If the total_increasing is removed, you also need to remove the `energy` class too avoid complaints from HA. Unfortunately it is not possible to implement a nett energy sensor that goes up and down without also having a last reset timestamp so HA can tell the difference between a drop due to excess solar production and a reset then consumption.
But since you have separate consumed and produced sensors, HA should be able to derive the nett use from those instead.

Reply:
That makes sense. Thank you. I've fixed it and it looks like it's working fine.